### ISSUE TITLE
First argument is lost when variadic arg is explicitly specified

### ISSUE THREAD

**Issue by Benjin** [yargs/yargs#1637]
```
let argv = require('yargs')
	.command('$0 [foods...]')
	.help().wrap(null).version(false).argv;

console.log(argv);
```

Arg not explicitly specified (good output):
```
PS F:\Source\scripts> node trial.js apples carrots
{ _: [], '$0': 'trial.js', foods: [ 'apples', 'carrots' ] }
```

Arg explicitly specified (bad output, expected to match above):
```
PS F:\Source\scripts> node trial.js --foods apples carrots
{ _: [], foods: [ 'carrots' ], '$0': 'trial.js' }
```

**Comment by mleguen** [comment]
Also note that:
```
PS F:\Source\scripts> node trial.js --foods apples --foods carrots
{ _: [], '$0': 'trial.js', foods: [ 'apples', 'carrots' ] }
```

So:
- either positionals should overwrite options, and `foods` should be empty here
- or they should be concatenated and `foods` should equal `[apple, carrots]` in @Benjin 's 2nd example

**Comment by bcoe** [comment]
thanks for the bug report, definitely seems like something we should address.

**Comment by Benjin** [comment]
> 
> 
> Also note that:
> 
> ```
> PS F:\Source\scripts> node trial.js --foods apples --foods carrots
> { _: [], '$0': 'trial.js', foods: [ 'apples', 'carrots' ] }
> ```
> 
> So:
> 
>     * either positionals should overwrite options, and `foods` should be empty here
> 
>     * or they should be concatenated and `foods` should equal `[apple, carrots]` in @Benjin 's 2nd example

Wouldn't "overwriting options" mean `foods` would just be `'carrots'` in that example?

Should it even be valid for a parameter to be specified twice?  I think I'd expect an error in that case, because otherwise we're guessing at the user's intention, as I think both concatenation and overwriting are equally iffy.

**Comment by jly36963** [comment]
I believe [my fix](https://github.com/yargs/yargs/pull/1992) was released, can you use yargs 17.3.1 and confirm that it solves the issue you reported?

The solution I chose is as follows:
- A user provides options and positionals under the same namespace
- If neither the option or the positional is array/variadic type, overwrite.
- If at least one is array/variadic, combine.

__NOTE__: The fix had a side-effect that concerns using array/variadic options/positionals in combination with `yargs.config()`. If you come across that behavior, I'm aware of it and currently working on a fix for it.

**Comment by shadowspawn** [comment]
This issue has been resolved. Thanks for comments, and PR from @jly36963
